### ISSUE TITLE
cdn_repo: make variants optional

### ISSUE THREAD

**Issue by ktdreyer** [ktdreyer/errata-tool-ansible#157]
Currently the `errata_tool_cdn_repo` module has a required `variants` parameter. This makes it difficult for products to start using errata-tool-ansible to manage CDN repos, because they have to specify all the variants up front. If they have do not list all variants with Ansible, this is a problem, and they cannot use `errata_tool_cdn_repo` at all.

Make the `variants` parameter optional, and update it to accept a "null" value:

a) If a user sets `variants: null` (the default for omitting the parameter): For entirely new CDN repos, this means "let the server decide the default value" (need to test that this works fine on a live ET instance). For existing resources, this means "don't change the list of variants on the server".

B) If a user sets `variants: []`: This means "remove all variants for this CDN repo"

C) If a user sets a non-empty list, like `variants: ["8Base-RHCEPH-4.0-Tools", "8Base-RHCEPH-4.1-Tools"]`: This behavior is obvious; simply set the variants on the server for new and existing CDN repos.

**Comment by hongweiliu17** [comment]
Thanks for the new solution!
In ET, variant is required when creating a new cdn_repo via restapi, otherwise it will prompt one error as below:
{"error":{"variant":["can't be blank"]}}

We must handle it.

**Comment by hongweiliu17** [comment]
Thanks for the new solution!
In ET, variant is required when creating a new cdn_repo via restapi, otherwise it will prompt one error as below:
{"error":{"variant":["can't be blank"]}}

We must handle it.

**Comment by ktdreyer** [comment]
I will play around with this API a bit more. One idea I had was that we could require a `variants` list on creation, and make it optional for editing.

**Comment by ktdreyer** [comment]
Yeah, the HTML UI requires one variant when we create the CDN repo.
